services/spar/test-integration/Test/Spar/APISpec.hs

```diff
@@ -268,12 +269,25 @@ specFinalizeLogin = do
               authnreq
               True
         sparresp <- submitAuthnResponse authnresp
+        let shouldContainInBase64 :: String -> String -> Expectation
+            shouldContainInBase64 hay needle = cs hay'' `shouldContain` needle
```

Comment:
This is already merged, but out of curiosity: Why do we search for the needle in the decoded hay, instead of searching for the encoded needle in the normal hay?

This would avoid the whole complication with finding the section between `<pre></pre>`.

Reply:
good idea.  just didn't occur to me.

but don't the character bounderies slide over the bits of the base64 encoding?  that would mean that I can't just encode the substring and search it in the encoding, because the bytes in the encoding depend on how many decoded bytes come before that substring.

Reply:
But wouldn't you get the same issue the other way around?

Reply:
I don't understand.  Can you elaborate?

This is the problem: the encoding of `"wef"` depends on what comes before it:

```
$ echo "wef" | base64 
d2VmCg==
$ echo "xwef" | base64 
eHdlZgo=
```

Reply:
Ah, yeah, it's because we not only encode the input, but `(err, inputs, mcky)`. Cool, makes sense. Thanks!